frontend/src/scenes/insights/InsightTooltip/insightTooltipUtils.tsx

```diff
@@ -0,0 +1,109 @@
+import { dayjs } from 'lib/dayjs'
+import React from 'react'
+import { ActionFilter, CompareLabelType } from '~/types'
+import { Space, Tag, Typography } from 'antd'
+import { capitalizeFirstLetter, midEllipsis, pluralize } from 'lib/utils'
+
+export interface SeriesDatum {
+    id: number // determines order that series will be displayed in
+    dataIndex: number
+    datasetIndex: number
+    breakdown_value?: string | number
+    compare_label?: CompareLabelType
+    action?: ActionFilter
+    label?: string
+    dotted?: boolean
+    color?: string
+    count: number
+}
+
+// Describes the row-by-row data for insight tooltips in the situation where series
+// are itemized as columns instead of rows by default
+export interface InvertedSeriesDatum {
+    id: string
+    datasetIndex: number
+    color?: string
+    datumTitle: React.ReactNode
+    seriesData: SeriesDatum[]
+}
+
+export interface InsightTooltipProps {
+    date?: string
+    hideInspectActorsSection?: boolean
+    seriesData?: SeriesDatum[]
+    tooltipAltTitle?: string | ((tooltipData: SeriesDatum[]) => React.ReactNode)
+    hideColorCol?: boolean
+    forceEntitiesAsColumns?: boolean
+    rowCutoff?: number
+    colCutoff?: number
+}
+
+export const COL_CUTOFF = 4
+export const ROW_CUTOFF = 8
+
+export function getTooltipTitle({ date, seriesData = [], tooltipAltTitle }: InsightTooltipProps): React.ReactNode {
+    // Use tooltip alternate title (or generate one if it's a function). Else default to date.
+    if (tooltipAltTitle) {
+        if (typeof tooltipAltTitle === 'function') {
+            return tooltipAltTitle(seriesData)
+        }
+        return tooltipAltTitle
+    }
+    return getFormattedDate(date)
```

Comment:
This works for most cases BUT when grouping by hour it doesn't show the relevant hour – I think it really should.